app/services/umrdr/uniqname_ldap_roles.rb

```diff
@@ -0,0 +1,76 @@
+module Umrdr
+
+  require 'rubygems'
+  require 'net/ldap'
+
+# For local development you can use the https://www.incommon.org/cert/repository/InCommonServerCA.txt
+# as a certiticate for verifying the MCommunity LDAP server's certificate.
+# ENV['LDAPTLS_CACERT'] = '/www/www.lib/misc/bundle.crt';
+
+  class UniqnameLdapRoles
+
+    attr_accessor :uniqname, :role_is_good
+
+    HOST = 'ldap.umich.edu'
+    PORT = 636
+    ENCRYPTION = :simple_tls
+    USERNAME =  'mcommunityuser'
+    PASSWORD = 'mcommunitypass'
+    AUTH = {:method => :simple, :username => USERNAME, :password => PASSWORD}
+
+    def self.uniqname_fac_or_staff(uname)
```

Comment:
Likely we'll never call `run` directly.  Although, you are correct and that could have a more meaningful name.

he convenience class method is likely the only function to be invoked in other code.